fluent-plugin-events/lib/fluent/plugin/in_events.rb

```diff
@@ -32,11 +32,17 @@ class EventsInput < Fluent::Plugin::Input
       config_param :namespace, :string, default: nil
       config_param :label_selector, :string, default: nil
       config_param :field_selector, :string, default: nil
+      config_param :type_selector, :array, default: ["ADDED", "MODIFIED"], value_type: :string
       config_param :configmap_update_interval_seconds, :integer, default: 10
       config_param :watch_interval_seconds, :integer, default: 300
       
       def configure(conf)
         super
+
+        @valid_types = ["ADDED", "MODIFIED", "DELETED"]
```

Comment:
Is this a complete list of the event types? I couldn't find any docs on this, would appreciate if you could point me in the right direction

Reply:
that reminds me there is also an `ERROR` type. I think it's errors related to using the watch API. in that case there won't be a resource version returned, and when that happens we skip the event and start a new watch stream. I will try to find docs on this, but I haven't seen any TBH

Reply:
[This](https://github.com/kubernetes/kubernetes/commit/2a76b82394a8853298f64bfe83df1015a05259e6#diff-54aa255ae082627d581223e1b6257a1aR44) code snippet shows the types. `BOOKMARK` is very interesting that I just discovered. It's an event that only contains a resource version for a resource. It acts like a heartbeat for watch that tells you even though you didn't get a watch event for a resource, there hasn't been any changes to that resource up to this resource version. more details [here](https://github.com/kubernetes/kubernetes/issues/73585). But it is still in alpha so I don't think we want to add it here yet.

Reply:
Thanks for looking into it Maisie, that sounds good to me.